projects/observability/src/pages/saved-queries/saved-queries.component.ts

```diff
@@ -0,0 +1,52 @@
+import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { NavigationService, PreferenceService } from '@hypertrace/common';
+import { DrilldownFilter, ExplorerService } from '../explorer/explorer-service';
+import { SavedQuery } from '../explorer/explorer.component';
+
+@Component({
+  styleUrls: ['./saved-queries.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="saved-queries">
+      <ht-page-header class="explorer-header"></ht-page-header>
+      <div class="query-list-container">
+        <a *ngFor="let query of savedQueries" (click)="onClick(query)">
+          <span *ngFor="let filter of query.filters">{{ filter.userString }}</span>
+        </a>
+      </div>
+      <p class="not-found-text" *ngIf="savedQueries.length === 0">
+        You haven't saved any queries! Go to Explorer page to save a query.
+      </p>
+    </div>
+  `
+})
+export class SavedQueriesComponent implements OnDestroy {
+  public savedQueries: SavedQuery[] = [];
+  private readonly subscriptions: Subscription = new Subscription();
+
+  public constructor(
+    private readonly navigationService: NavigationService,
+    private readonly preferenceService: PreferenceService,
+    private readonly explorerService: ExplorerService
+  ) {
+    this.subscriptions.add(
+      this.preferenceService.get('savedQueries', []).subscribe(queries => {
+        this.savedQueries = queries as SavedQuery[];
+      })
+    );
+  }
```

Comment:
nit - You can make use of `async` or `ht-async` pipe here. So, what you can do is like follows
```
this.savedQueries$ = this.preferenceService.get('savedQueries', []);
```
In the html part
```
*ngFor = let query of this.savedQueries$ | async
```

And same for length too
```
*ngIf="(this.savedQueries$ | async)?.length === 0"
```

I think it might make the code a bit more cleaner.

Reply:
Worked like a charm, thanks! Done ✅